### ISSUE TITLE
Snapshot show up as protected without using a password

### ISSUE THREAD

**Issue by chanz** [Sebclem/hassio-nextcloud-backup#70]
**Describe the bug**
I got no password set in settings > backup > security.
Yet restoring snapshots is not possible due to a wrong password. Also snapshots show up as "protected" = "true".

**To Reproduce**
Disable password protection for snapshos and manually trigger a snapshot within NC Backup.

**Expected behavior**
Be able to restore snapshots without a password and snapshots show as "protected" = "false".

**System information**
- Home Assistant Version : Home Assistant OS 5.13
- Home Assistant Supervisor Version : supervisor-2021.05.4
- Nextcloud Version : 21
- Addon version : 0.16.1 
- If any, reverse proxy in fort on Nextcloud : none

**Additional context**
What password is currently being used? I mean I set no password, yet its still encrypted?

**Comment by Sebclem** [comment]
Did the you turn off the `password protected` switch ? (Just on top of the password field)
With this switch off I have all my backup as expected (`protected=flase`)

But I will add a check to prevent issue : if `password protected` is on, check is password is not empty.

I will add this as soon as I can, I'm a little busy this month.

**Comment by chanz** [comment]
Well currently my settings look like in the following screenshot. But well Snapshots still show up as protected and I can't restore them.

![Screenshot_20210606_002515](https://user-images.githubusercontent.com/1131900/120906954-e586b180-c65d-11eb-9d7c-15a182feec84.png)

![Screenshot_20210606_002807](https://user-images.githubusercontent.com/1131900/120907001-30082e00-c65e-11eb-8b04-8d9edc399375.png)

**Comment by Sebclem** [comment]
Very strange, I will try to get some time tomorrow to dig into this issue (It's 1:00AM here 😅)
But my first idea a corrupted setting file but I don't know how it's possible.
I will dig into this.

**Comment by chanz** [comment]
Thank you very much. Let me know if I can provide any thing else.

**Comment by Sebclem** [comment]
Do you have any warning in the addon logs when you save the config ?

**Comment by chanz** [comment]
With loglevel info I just see the following, when saving the settings:
```
[2021-06-08 00:24:45] [info]: 	Stopping Cron...
[2021-06-08 00:24:45] [info]: 	Starting Cron...
```
Using debug I can see the following:
```
[2021-06-08 00:26:44] [debug]: 	exclude_folder is not array (Empty value), reset...
[2021-06-08 00:26:44] [debug]: 	exclude_addon is not array (Empty value), reset...
[2021-06-08 00:26:44] [info]: 	Stopping Cron...
[2021-06-08 00:26:44] [info]: 	Starting Cron...
```

To test it more, I turned password protection on. But the settings page sometimes forgets my password and turns password protection off. I try to find a cause for you to reproduce, but found nothing so far.

**Comment by Sebclem** [comment]
Hum very strange, I will take a look if nothing is wrong with the settings initialization

**Comment by Sebclem** [comment]
Nop, I still need to take a look at this :sweat_smile:

**Comment by chanz** [comment]
<3
Since it works now, after I turned "Passwort Protection" on and off and restarting in between, I fear I can't be assisting you right now. But let me know how I can buy you a beer! :-)

**Comment by xxgmxx** [comment]
I am in a situation where I cannot restore backups because all of them are password-protected. I know that I got the password protection off but it wants me to put a password in while restoring. My HA is dead and all I have is password protected backups I cant restore :/ . Is it possible that the blank password put something in, like a default password?

**Comment by Sebclem** [comment]
Arfff, I don't understand this bug...

You can try with "null" or "undefined" but this bug is very strange.
I'm sorry if this doesn't work... I have reviewed all my code and I didn't find anything...

**Comment by xxgmxx** [comment]
I think that it kind of remembers the password, but shows like there is none. I just started a new VM to test it, and tried that again. When I checked the "password protected" and put the password there, saved and made a backup, then the backup is really protected by the password, but the addon shows that there is no password and even that the password protection is off. I did tried to protect my backups but it always unchecked the password protected to off, but seems like it actually works just do not show up correctly.
